src/usr/local/kobocloud/getRemoteFile.sh

```diff
@@ -2,39 +2,41 @@
 
 linkLine="$1"
 localFile="$2"
-user="$3"    
+user="$3"
+outputFileTmp="/tmp/kobo-remote-file-tmp.log"
 
 #load config
 . `dirname $0`/config.sh
 
-
-if [ "$user" = "" ]; then
-    curlCommand=$CURL
-else
-    curlCommand="$CURL -u $user: "
-fi
-    
-#echo $curlCommand
-    
-echo "$linkLine -> $localFile"
-
-remoteSize=`$curlCommand -k -L --silent --head "$linkLine" | tr A-Z a-z | sed -n 's/^content-length\: \([0-9]*\).*/\1/p'`
-echo "Remote size: $remoteSize"
-if [ -f $localFile ]; then
-  localSize=`stat -c%s "$localFile"`
-else
-  localSize=0
+curlCommand="$CURL"
+if [ ! -z "$user" ] && [ "$user" != "-" ]; then
+    echo "User: $user"
+    curlCommand="$curlCommand -u $user: "
 fi
-if [ "$remoteSize" = "" ]; then
-  remoteSize=1
+
+echo "Download: "$curlCommand -k --silent -C - -L -o "$localFile" "$linkLine" -v
+
+$curlCommand -k --silent -C - -L -o "$localFile" "$linkLine" -v 2>$outputFileTmp
+status=$?
+echo "Status: $status"
+echo "Output: "
+cat $outputFileTmp
+
+statusCode=`cat $outputFileTmp | grep 'HTTP/' | tail -n 1 | cut -d' ' -f3`
+rm $outputFileTmp
+
+echo "Remote file information:"
+echo "  Status code: $statusCode"
+
+if echo "$statusCode" | grep -q "403"; then
+    echo "Error: Forbidden"
+    rm $localFile
```

Comment:
In the case we got a error we delete the file. What can happen is that the file was correctly downloaded on a previous sync, and then we delete on a subsequent sync when the service is returning errors.

To avoid removing everything this way the GDriveFiles check the exit code to stop looping over the rest of the files. All implementations must have it.